src/cryptonote_core/service_node_list.cpp

```diff
@@ -1802,62 +1819,46 @@ namespace service_nodes
       return false;
     }
 
+    // NOTE: Deserialize long term state history
+    uint64_t bytes_loaded = 0;
     cryptonote::db_rtxn_guard txn_guard(m_db);
     std::string blob;
-    if (!m_db->get_service_node_data(blob))
+    if (m_db->get_service_node_data(blob, true /*long_term*/))
     {
-      return false;
-    }
-
-    std::stringstream ss;
-    ss << blob;
-    binary_archive<false> ba(ss);
+      bytes_loaded += blob.size();
+      LOKI_DEFER { blob.clear(); };
+      std::stringstream ss;
+      ss << blob;
+      binary_archive<false> ba(ss);
```

Comment:
This is a task for another time (basically just leaving a note for myself here): it's fairly easy to write an in-place input stream buffer that can avoid the need for copying a temporary string into a stringstream.